### ISSUE TITLE
Temporary files created with insufficient permissions prevents preview re-rendering

### ISSUE THREAD

**Issue by odrotbohm** [de-jcup/eclipse-asciidoctor-editor#412]
I am using the just release Asciidoctor Editor 2.5 on MacOS. Steps to reproduce:

1. Open an Asciidoctor file in Eclipse.

The editor bootstraps the ASP and creates a couple of files in some `/var/folders/…` folder:

```bash
$ ls -la
total 32
drwxr-xr-x  5 odrotbohm  staff   160 May  2 16:48 .
drwxr-xr-x  3 odrotbohm  staff    96 May  2 16:42 ..
-rw-r--r--  1 odrotbohm  staff   481 May  2 16:48 2e6e64b67c06ccd5f74d09d5512d553d092ac73ec879dbc4082a87c17cf1efee_hidden-editorfile_….adoc
-r----x---  1 odrotbohm  staff  2830 May  2 16:48 2e6e64b67c06ccd5f74d09d5512d553d092ac73ec879dbc4082a87c17cf1efee_hidden-editorfile_….html
-rw-r--r--  1 odrotbohm  staff  4124 May  2 16:48 2e6e64b67c06ccd5f74d09d5512d553d092ac73ec879dbc4082a87c17cf1efee_internal_….html
```

Note how the hidden editorfile is created without write permissions even for the owner. This causes the previe to render fine but then no edits actually causing the preview being updated as the generation fails with (from the Eclipse console):

```
ASP: Processing file:/Users/….adoc
Command convert_file failed.Failed:org.jruby.exceptions.SystemCallError: (EACCES) Permission denied - /var/folders/…/2e6e64b67c06ccd5f74d09d5512d553d092ac73ec879dbc4082a87c17cf1efee_hidden-editorfile_….html
```

**Comment by odrotbohm** [comment]
Works fine if I switch to using a locally installed Asciidoctor (via preferences).

**Comment by de-jcup** [comment]
Thanx for reporting - will try to fix it ASAP

**Comment by de-jcup** [comment]
There is currently no explicit file permission handling defined with #417 this will be done . So only the user creating/generating the document shall be able to open, read and write the document. This should fix the mentioned problem as well.

**Comment by de-jcup** [comment]
With #417 and #434  the temp data for the editor is no longer in system temp folder but in a dedicated tmp folder in user.home. 
So the problem should be fixed when Version 2.6.0 is deployed on eclipse marketplace.

**Comment by de-jcup** [comment]
@odrotbohm : I just released Version 2.6.0 on eclipse marketplace. Could you please do me a favor and check if the new version has no longer this problem on a MAC?

**Comment by odrotbohm** [comment]
I see the rendering not fail with original error anymore but file-relative inclusions are now broken. I have my source file `index.adoc` in `src/main/asciidoc`. I set an attribute `:mybase:` to `../../../target/generated-docs` and then use e.g. `plantuml::{mybase}/components.uml`. The console then has the following:

```
Installed asciidoctor: Processing file:/…/src/main/asciidoc/index.adoc
asciidoctor: FAILED: /Users/odrotbohm/.eclipse-asciidoctor-editor/tmp/spring-io-2022/822ad436-fbd5-3fcc-a267-9ee3a36061e8_edifile_h_index.adoc: Failed to load AsciiDoc document - No such file or directory @ rb_sysopen - /target/generated-docs/components.uml
  Use --trace to show backtrace
```

which is not wrong per se because at the location the source file is copied to that relative file doesn't exist, but I guess exactly that copying is asking for all kinds of trouble unless you move all related resources that could be referred to to that place as well. Is there a particular reason you copy the files before rendering them? Shouldn't it be fine to just direct the output of the rendering to a dedicated temp folder?

**Comment by de-jcup** [comment]
@odrotbohm : I am sorry - I have currently exact the same problems and I created #438  to fix this. The base directory for the diagrams is no longer correct calculated (I deleted my former comment to you because of the bug... but of course you received a mail... sorry)

**Comment by de-jcup** [comment]
@odrotbohm : I just released Version 2.6.1 to eclipse marketplace. It works with my documentation project + diagrams etc. So please try again.

**Comment by odrotbohm** [comment]
No need to apologize at all! You're doing remarkable work for the Asciidoctor community. Highly appreciated! I was just happy I could provide feedback 🙃.

Even better, I can report the previews now rendering as expected! Thanks again!
